test/Microsoft.AspNetCore.Authentication.Test/Cookies/CookieMiddlewareTests.cs

```diff
@@ -739,11 +739,19 @@ private Task SignOutAsWrong(HttpContext context)
             Assert.Equal(HttpStatusCode.Redirect, transaction2.Response.StatusCode);
         }
 
-        [Fact]
-        public async Task MapWillAffectChallenge()
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task MapWillAffectChallengeOnlyWithUseAuth(bool useAuth)
```

Comment:
FYI @PinpointTownes fixed the map + resolve url behavior to work similar to before (using the path of the single middleware), via a new IAuthenticationFeature like we discussed @Tratcher